workflow/rules/ref.smk

```diff
@@ -35,13 +37,15 @@ rule extract_genome:
         "../envs/reference.yml"
     shell:
         """
-        unzip -p {input} ncbi_dataset/data/{params.accession}/*.fna > {output} 2> {log}
+        [ -f "{input}" ] && mv {input} {output} || unzip -p {input} ncbi_dataset/data/{params.accession}/*.fna > {output} 2>> {log}
         """
 
 
 rule extract_annotation:
     input:
-        rules.get_genome.output,
+        lambda wildcards: get_reference_files(config).get(
+            "annotation", "references/ncbi_dataset.zip"
+        ),
```

Comment:
@coderabbit - yes, placing shell variables in quotation marks will avoid string splitting (even though it is not likely to occur because file names are checked beforehand). But anything else is not necessary. This is part of a Sakemake workflow and validation of files occurred, except for the gff file format - the aim of this rule is to ensure its correctness by extracting it from a reference OR using a supplied gff, then downstream the integrity is ensured.